contract-ui/tabs/overview/helpers/getGuidesAndTemplates.ts

```diff
@@ -0,0 +1,386 @@
+  },
+  dao: {
+    title: "DAO",
+    url: "https://github.com/thirdweb-example/dao",
+  },
+  multiwrap: {
+    title: "Multiwrap",
+    url: "https://github.com/thirdweb-example/multiwrap",
+  },
+};
+
+const TEMPLATES_FOR_TYPE: { [key: string]: Item[] } = {
+  pack: [ALL_TEMPLATES.packs],
+  "signature-drop": [ALL_TEMPLATES.signatureDrop],
+  marketplace: [ALL_TEMPLATES.marketplace],
+  vote: [ALL_TEMPLATES.dao],
+  multiwrap: [ALL_TEMPLATES.multiwrap],
+  "nft-drop": [ALL_TEMPLATES.nftDrop],
+  "edition-drop": [ALL_TEMPLATES.editionDrop, ALL_TEMPLATES.nftGatedWebsite],
+  "token-drop": [ALL_TEMPLATES.tokenDrop],
+};
+
+const TEMPLATES_FOR_FEATURE: { [key: string]: Item[] } = {
+  ERC20ClaimConditionsV1: [ALL_TEMPLATES.tokenDrop],
+  ERC20ClaimConditionsV2: [ALL_TEMPLATES.tokenDrop],
+  ERC20ClaimPhasesV1: [ALL_TEMPLATES.tokenDrop],
+  ERC20ClaimPhasesV2: [ALL_TEMPLATES.tokenDrop],
+  ERC20Burnable: [],
+  ERC20Revealable: [],
+  ERC20TieredDrop: [],
+  ERC20ClaimCustom: [],
+  ERC20LazyMintable: [],
+  ERC20BatchMintable: [],
+  ERC20Mintable: [],
+  ERC20: [ALL_TEMPLATES.nftStakingApp],
+  ERC721Burnable: [],
+  ERC721Revealable: [],
+  ERC721TieredDrop: [],
+  ERC721ClaimConditionsV1: [ALL_TEMPLATES.nftDrop],
+  ERC721ClaimConditionsV2: [ALL_TEMPLATES.nftDrop],
+  ERC721ClaimPhasesV1: [ALL_TEMPLATES.nftDrop],
+  ERC721ClaimPhasesV2: [ALL_TEMPLATES.nftDrop],
+  ERC721ClaimCustom: [],
+  ERC721LazyMintable: [ALL_TEMPLATES.burn1155Mint721],
+  ERC721BatchMintable: [],
+  ERC721Mintable: [],
+  ERC721SignatureMintV2: [ALL_TEMPLATES.signatureBasedMinting],
+  ERC721SignatureMintV1: [ALL_TEMPLATES.signatureBasedMinting],
+  ERC721Enumerable: [],
+  ERC721Supply: [],
+  ERC721: [ALL_TEMPLATES.nftGatedWebsite],
+  ERC1155Burnable: [],
+  ERC1155ClaimConditionsV1: [ALL_TEMPLATES.editionDrop],
+  ERC1155ClaimConditionsV2: [ALL_TEMPLATES.editionDrop],
+  ERC1155ClaimPhasesV2: [ALL_TEMPLATES.editionDrop],
+  ERC1155ClaimPhasesV1: [ALL_TEMPLATES.editionDrop],
+  ERC1155ClaimCustom: [],
+  ERC1155Revealable: [],
+  ERC1155LazyMintableV2: [],
+  ERC1155LazyMintableV1: [],
+  ERC1155SignatureMintable: [],
+  ERC1155BatchMintable: [],
+  ERC1155Mintable: [],
+  ERC1155Enumerable: [],
+  ERC1155: [ALL_TEMPLATES.nftGatedWebsite],
+  Royalty: [],
+  PrimarySale: [],
+  PlatformFee: [],
+  PermissionsEnumerable: [],
+  Permissions: [],
+  ContractMetadata: [],
+  AppURI: [],
+  Ownable: [],
+};
+
+export function getGuidesAndTemplates(
+  features: string[],
+  contractType: ContractType,
+): {
+  guides: Item[];
+  templates: Item[];
+} {
+  const data = features.reduce(
+    (acc, guide) => ({
+      guides: [...acc.guides, ...(GUIDES_FOR_FEATURE[guide] || [])],
+      templates: [...acc.templates, ...(TEMPLATES_FOR_FEATURE[guide] || [])],
+    }),
+    { guides: [] as Item[], templates: [] as Item[] },
+  );
+
+  data.guides.push(...(GUIDES_FOR_TYPE[contractType] || []));
+  data.templates.push(...(TEMPLATES_FOR_TYPE[contractType] || []));
```

Comment:
prefer `.concat()` and using the returned data instead of mutating the array

also we probably want to put the "for contract type" guides and templates *first* in the array (pack contract type base guide should be above nft feature-basedguides)